### ISSUE TITLE
bug: Wan2.2-TI2V-5B all-NaN at 1280x704 (seq=27280 self-attn overflow) — static video output

### ISSUE THREAD

**Issue by dahai80** [dahai80/fusion-mlx#500]
## Summary

`Wan2.2-TI2V-5B` (mlx-q8, `~/.fusion-mlx/models/Wan2.2-TI2V-5B-mlx-q8`) produces an **all-static video** (all 121 frames identical, `frame_diffs=[0.0,0.0]`) when run at **1280x704 x 121 frames** via the ComfyUI `video_wan2_2_5B_ti2v` acceptance workflow. The diffusion denoising overflows to **all-NaN** latents, and the VAE decode replaces NaN with 0 → frames collapse.

## Reproduction

Direct `generate_video` call (no ComfyUI needed):

```python
from fusion_mlx.video.wan2.generate import generate_video
generate_video(
    model_dir="~/.fusion-mlx/models/Wan2.2-TI2V-5B-mlx-q8",
    prompt="<long 353-token prompt>", negative_prompt="<zh neg>",
    image="<start.png>", width=1280, height=704,
    num_frames=121, steps=4, guide_scale=5.0, shift=5.0,
    seed=2524546589, scheduler="unipc", no_compile=True,
    debug_latents=True, output_path="/tmp/ti2v_debug.mp4",
)
```

### Root-cause isolation (debug_latents diagnostics)

| Config | seq_len | Step-1 latents | Output |
|---|---|---|---|
| 17f x 512x512 | 1280 | all positions distinct, no NaN | **PASS** (frame diffs 27-50) |
| 121f x 512x512 | ~3200 | all positions distinct, no NaN | **PASS** |
| 121f x 1280x704 | 27280 | `Denoise step 1/4: 5237568 NaN in latents — aborting` | **FAIL** all-NaN |

So the trigger is **resolution 1280x704** (seq=27280), NOT the frame count (121f works fine at 512x512).

`FUSION_WAN2_ATTN_CHUNK=8192` does **not** fix it — latents still go all-NaN.

## Relevant logs

```
[WARNING] wan2 self-attn seq=27280 exceeds safe threshold; set FUSION_WAN2_ATTN_CHUNK (e.g. 8192) to avoid Metal freeze
Denoise step 1/4: 5237568 NaN in latents — aborting
Wan2 VAE decode (wan22): 327106560 NaN in video, replacing with 0
Latent diagnostics: Pos 0..7 all nan
```

## Suspected cause

The `ti2v` model_type takes the `is_i2v_mask_blend` path (generate.py:628, model_type != "i2v"). The single-model forward at seq=27280 overflows in bfloat16/fp8_e4m3 attention (RoPE or softmax) → NaN propagates through the mask-blend `latents = (1-mask)*z_img + mask*latents` (generate.py:1022,1154) to all temporal positions, then VAE decode zeros them.

The 1.3B `fun_camera` model at the same resolution (seq=21504) does NOT NaN (just slow) — so the 5B ti2v is more numerically fragile than 1.3B at large seq.

## Workaround

Downscale to 512x512: produces correct motion. But the official ComfyUI workflow ships at 1280x704.

## Expected

Either (a) auto-cap/clamp seq for ti2v-5B to avoid overflow, (b) fix the attention/RoPE numerical path for large seq, or (c) detect NaN and fall back to a safe resolution. The `FUSION_WAN2_ATTN_CHUNK` env should ideally prevent this but does not.

cc @dahai80

**Comment by dahai80** [comment]
## Root cause localized: q8 quantized matmul bf16-accumulation overflow

Update from deeper instrumentation. Added per-block NaN probes inside `WanAttentionBlock.__call__` + `WanCrossAttention.__call__` (diagnostic, reverted after). Ran 1280x704 x121f, 1 step, `FUSION_WAN2_ATTN_CHUNK=8192`:

### Findings
1. `z_img` (VAE encode of start image) is **finite** — mean=0.0202 std=0.3093 nan=0. VAE encode is not the cause.
2. `noise_pred` is **all-NaN at step 0** — the model forward itself produces NaN (not the scheduler).
3. Block 0 is **clean** (no NaN anywhere in block 0).
4. **First NaN: block 1, `post_ca`** (after cross-attention), nan=64145/167608320 (0.04%). Then propagates and explodes through blocks 2+.
5. Inside cross-attn, the q-projection overflows:

```
CROSS q_in:   absmax=41.0706  nan=0   dtype=float32   # block 0 cross-attn input (finite!)
CROSS q_proj: absmax=nan      nan=61031               # self.q(x) -> NaN from finite input
```

### Root cause
The **cross-attention q-projection** `self.q(x)` (a q8 `QuantizedLinear`) overflows when the **un-normalized residual stream** `x` has large magnitude (absmax=41 at 1280x704). Self-attention is protected by `norm1(x)` before its q-projection; cross-attention applies `self.q(x)` **directly on the residual** (no pre-norm when `cross_attn_norm=False`).

The overflow is **bf16 accumulation** in the q8 matmul: input absmax=41 × dim=5120 partial sums ≈ 41×5120 ≈ 210000, exceeding bf16 max (65504) → inf → NaN. This is resolution-dependent because the residual stream magnitude grows with seq length / spatial dims.

### Partial fix tested (NOT sufficient)
Upcasting the cross-attn q-projection to fp32 (`_inner_q(x.astype(mx.float32))` for `QuantizedLinear`) **fixed block 0** (`q_proj absmax=12.10, nan=0`). But NaN then appeared at **block 0 `ffn_out`** and block 1 self-attn — the same q8 accumulation overflow affects **all quantized linears** once the residual stream grows, not just cross-attn q.

### Conclusion
This is a **fundamental q8 quantization dynamic-range issue** at high resolution (seq≥~27k): the residual stream magnitude exceeds what q8 matmuls with bf16 accumulation can represent. The reference PyTorch path uses bf16 autocast (no quantization, range ±65504 throughout) and does not overflow.

### Fix directions (need investigation)
- (a) Force **fp32 accumulation** for q8 matmuls on large-magnitude inputs (may require MLX kernel support / `mx.quantized_matmul` with higher-precision accumulate — need to check if MLX exposes this).
- (b) Dequantize overflow-prone linears to fp32/bf16 at high seq (correctness win, perf/memory cost).
- (c) Manage residual stream magnitude (e.g. clamp/scale before quantized matmuls) — risks output drift.
- (d) Gate the model to a max-resolution / max-seq where q8 stays in range (workaround, not a fix).

512x512 (seq~3200) and 17f work because the residual magnitude stays within q8/bf16 range. This is **not** an attention-matrix-size issue (attn chunking doesn't help) — it's the **linear-projection matmul accumulation** dtype.

Repro script (recreated as needed):
```python
generate_video(model_dir="~/.fusion-mlx/models/Wan2.2-TI2V-5B-mlx-q8",
  prompt="...", negative_prompt="...", image="<start.png>",
  width=1280, height=704, num_frames=121, steps=1, guide_scale=5.0,
  shift=5.0, seed=2524546589, scheduler="unipc", no_compile=True,
  debug_latents=True, output_path="/tmp/ti2v_debug.mp4")
```
